javascript/javascript3/week3/promise-classmates.js

```diff
@@ -0,0 +1,47 @@
+//Getting into promises
+
+
+function promiseToClassmates(userName){
+    return new Promise((resolve, reject) => {
+        fetch(`https://api.github.com/search/repositories?q=user:${userName}`)
+        .then(res=>res.json()) 
+        .then(data=>{
+            console.log(data);
+            renderGitRepoData(data, userName);
+            console.log("all promised has resolved")
+        resolve(data)})
+        .catch((e)=> reject(e))
+    })
+}
+function renderGitRepoData(data, userName){
+    const ulName = document.querySelector("ul");
+    const li = document.createElement("li")
+    ulName.appendChild(li);
+    li.innerHTML = `${userName} repositories`;
+    const ulRepo = document.createElement("ul")
+    li.appendChild(ulRepo);
+    for(let i = 0; i < 3; i++){
+        const li = document.createElement("li");
+        li.setAttribute("class", "url");
+        li.innerHTML = `${data.items[i].name}: <a href="#"> ${data.items[i].git_url} </a>`
+        ulRepo.appendChild(li);
+    }
+}
```

Comment:
You don't render the data that the task is asking for. The username is not that important here, I would remove it from the function parameters.
Items is an array of objects that holds information about each repository. So you can just loop through this array and retrieve some information. Have a look here to get an understanding of the structure of the data:
https://api.github.com/search/repositories?q=user:benna100
So 
```
items.forEach(item => {
// Here you can access eg. item.full_name, item.url, item.owner.login
})
```
We want to render all the repositories regardless of whose, and for each repository we want display what is the full name of the repository(**item.full_name**), what is the URL of the repo(**item.url**), and who is the owner of the repository(**item.owner.login** gives you the github username).

Reply:
Its fixed.